### PR TITLE
feat(helm): prompt for password with helm repo add

### LINKED ISSUE #7174: helm3: repo add no longer interactively asks for password
Output of `helm version`: version.BuildInfo{Version:"v3.0.0", GitCommit:"e29ce2a54e96cd02ccfce88bee4f58bb6e2a28b6", GitTreeState:"clean", GoVersion:"go1.13.4"}

With helm v2, passing --username myname to helm repo add caused helm to interactively ask for a password which is positive for security reasons (not having a password show up in process lists and shell histories, the password being shadowed while entering).

helm v3 seems to have lost this ability, which looks like a regression to me.

### PR DESCRIPTION
**Pull request by jrweldon**
**What this PR does / why we need it**:
Closes #7174

Previously in Helm 2, `helm repo add` would prompt for your password if you only provided the `--username` flag. This helps prevent someone's credentials from being logged in their shell's history, from showing up in process lists, and from being seen by others nearby.

**Special notes for your reviewer**:
This is my first time contributing to a large open-source project, please let me know if anything else is needed or desired and I'll do my best to address the concerns as soon as possible.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/repo_add.go [hunk 1]
```diff
@@ -114,6 +116,16 @@ func (o *repoAddOptions) run(out io.Writer) error {
 		return errors.Errorf("repository name (%s) already exists, please specify a different name", o.name)
 	}
 
+	if o.username != "" && o.password == "" {
+		fmt.Fprint(out, "Password:")
```
Comment (bacongobbler) [inline cmd/helm/repo_add.go:120]: can you add a whitespace character after the colon character for formatting?

**Review by bacongobbler — CHANGES_REQUESTED** [review]
thanks for looking into this. I left one note for feedback.

Have you tested this yet to make sure it works?

File: cmd/helm/repo_add.go (hunk 1 above)
Comment (jrweldon) [reply cmd/helm/repo_add.go]: Sure thing, I just pushed a new commit with the space.

**Comment by jrweldon** [comment]
> thanks for looking into this. I left one note for feedback.
> 
> Have you tested this yet to make sure it works?

Yes, I have tested locally using a private repository. My `${XDG_CONFIG_HOME}/helm/repositories.yaml` was updated to include the new entry, similar to how Helm 2 behaved.

**Review by bacongobbler — APPROVED** [review]

**Comment by philomory** [comment]
Did this feature not make it into Helm 3.3.0?

**Comment by jrweldon** [comment]
No, it was a few days after the 3.3 cutoff, it'll be in 3.4

**Pull request merged**